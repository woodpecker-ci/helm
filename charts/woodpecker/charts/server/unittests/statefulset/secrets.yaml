# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: secrets
release:
  name: woodpecker-unittests
  namespace: testing
templates:
  - templates/secret.yaml
tests:
  - it: creates a default `woodpecker-secret` secret with default values
    values:
      - ../../values.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Secret
      - matchRegex:
          path: metadata.name
          pattern: woodpecker-secret
  - it: populates the custom secret values properly
    set:
      secrets:
        - name: woodpecker-secret
          data:
            FOO: bar
            BAZ: foo
    asserts:
      - matchSnapshot:
          path: data
  - it: can create an extra secret with proper values
    set:
      secrets:
        - name: woodpecker-secret
          data:
            FOO: bar
            BAZ: foo
        - name: extra-secret
          data:
            FOO: bar
            BAZ: foo
    documentSelector:
      path: metadata.name
      value: extra-secret
    asserts:
      - hasDocuments:
          count: 2
      - isKind:
          of: Secret
      - matchRegex:
          path: metadata.name
          pattern: extra-secret
      - matchSnapshot:
          path: data
---
suite: secret-check-job
release:
  name: woodpecker-unittests
  namespace: testing
templates:
  - templates/secret.yaml
tests:
  - it: creates a default `woodpecker-secret` secret with default values
    values:
      - ../../values.yaml
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Secret
      - matchRegex:
          path: metadata.name
          pattern: woodpecker-secret
  - it: populates the custom secret values properly
    set:
      secrets:
        - name: woodpecker-secret
          data:
            FOO: bar
            BAZ: foo
    asserts:
      - matchSnapshot:
          path: data
  - it: can create an extra secret with proper values
    set:
      secrets:
        - name: woodpecker-secret
          data:
            FOO: bar
            BAZ: foo
        - name: extra-secret
          data:
            FOO: bar
            BAZ: foo
    documentSelector:
      path: metadata.name
      value: extra-secret
    asserts:
      - hasDocuments:
          count: 2
      - isKind:
          of: Secret
      - matchRegex:
          path: metadata.name
          pattern: extra-secret
      - matchSnapshot:
          path: data
