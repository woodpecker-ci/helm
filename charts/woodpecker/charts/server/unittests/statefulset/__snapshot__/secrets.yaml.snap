can create an extra secret with proper values:
  1: |
    BAZ: Zm9v
    FOO: YmFy
populates the custom secret values properly:
  1: |
    BAZ: Zm9v
    FOO: YmFy
